### PR TITLE
chore: update default values in configs

### PR DESCRIPTION
**Pull request by mrehan27**
fixes: https://linear.app/customerio/issue/MBL-139/update-config-values

### Changes

- Removed `defaultSettings`, `flushQueue` and `operatingMode` from `SDKConfigBuilder` and `DataPipelineConfigOptions` completely
- Updated `autoAddCustomerIODestination` to `internal` as it is required to suppress requests in tests
- Updated tests and added a few more to improve verify expectations for analytics configuration

### CODE REVIEW

File: Sources/DataPipeline/Config/SDKConfigBuilder.swift [hunk 1]
```diff
@@ -119,35 +116,17 @@ public class SDKConfigBuilder {
     }
 
     @discardableResult
-    public func autoAddCustomerIODestination(_ autoAdd: Bool) -> SDKConfigBuilder {
+    func autoAddCustomerIODestination(_ autoAdd: Bool) -> SDKConfigBuilder {
```
Comment (mrehan27) [inline Sources/DataPipeline/Config/SDKConfigBuilder.swift:119]: Updated it to internal as it is required to suppress requests in tests

File: Tests/DataPipeline/APITest.swift [hunk 2]
```diff
@@ -127,11 +127,7 @@ class DataPipelineAPITest: UnitTest {
             .cdnHost("")
             .flushAt(10)
             .flushInterval(10.0)
-            .autoAddCustomerIODestination(false)
```
Comment (mrehan27) [inline Tests/DataPipeline/APITest.swift:130]: Removed it from here as it is no longer `public`

File: Tests/DataPipeline/Config/DataPipelineConfigOptionsTests.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
+@testable import CioDataPipelines
+@testable import Segment
+import SharedTests
+import XCTest
+
+// Tests to verify that configurations created by the SDK for our DataPipeline module and Segment analytics matches our expectations.
+class DataPipelineConfigOptionsTests: UnitTest {
```
Comment (mrehan27) [inline Tests/DataPipeline/Config/DataPipelineConfigOptionsTests.swift:7]: I added this to hold all config compatibility related tests at single place

File: Tests/DataPipeline/Config/SDKConfigBuilderTest.swift [hunk 4]
```diff
@@ -68,44 +59,13 @@ class SDKConfigBuilderTest: UnitTest {
...
         XCTAssertSame(dataPipelineConfig.flushPolicies, givenFlushPolicies)
-        XCTAssertSame(dataPipelineConfig.flushQueue, givenFlushQueue)
-        XCTAssertEqual(dataPipelineConfig.operatingMode, givenOperatingMode)
         XCTAssertEqual(dataPipelineConfig.trackApplicationLifecycleEvents, givenTrackApplicationLifecycleEvents)
         XCTAssertEqual(dataPipelineConfig.autoTrackDeviceAttributes, givenAutoTrackDeviceAttributes)
         XCTAssertEqual(dataPipelineConfig.migrationSiteId, givenSiteId)
         XCTAssertEqual(dataPipelineConfig.autoConfiguredPlugins.count, 0)
     }
 
-    func test_givenDefaults_expectMatchAnalyticsDefaults() {
```
Comment (mrehan27) [inline Tests/DataPipeline/Config/SDKConfigBuilderTest.swift:81]: This is moved to `DataPipelineConfigOptionsTests`

**Review by Shahroz16 — APPROVED** [review]
Changes look according to the table in the linear.

**Pull request merged**